ff_dir.c

```diff
@@ -1723,9 +1723,11 @@ FF_Error_t FF_PopulateLongDirent( FF_IOManager_t * pxIOManager,
                     {
                         /* Valid Dir found, copy the wildCard to filename! */
                         #if ( ffconfigUNICODE_UTF16_SUPPORT != 0 )
-                            wcsncpy( pxDirEntry->pcWildCard, ++pcWildCard, ffconfigMAX_FILENAME );
+                            wcsncpy( pxDirEntry->pcWildCard, ++pcWildCard, ffconfigMAX_FILENAME - 1 );
+                            pxDirEntry->pcWildCard[ sizeof( pxDirEntry->pcWildCard ) - 1 ] = 0;
```

Comment:
You wrote:

> `pxDirEntry->pcWildCard[ sizeof( pxDirEntry->pcWildCard ) - 1 ] = 0;`

Now that I see the new code, I realise that `sizeof( pcWildCard  )` isn't equal to the number of elements in an array of `wchar_t`.
I think that we should write:
~~~diff
-  pxDirEntry->pcWildCard[ sizeof( pxDirEntry->pcWildCard ) - 1 ] = 0;
+  pxDirEntry->pcWildCard[ ffconfigMAX_FILENAME ] = 0;
~~~
In ff_dir.c, `strncpy()` and `wcsncpy()` are called, depending on `ffconfigUNICODE_UTF16_SUPPORT`, whereas in ff_file.c, the macro `STRNCPY` is used.
What about having ff_dir.c use that macro as well?

PS. My proposal to use `sizeof` as an index is that it is never mistaken about the actual length of a string. Except when it is a string of 16-bit entries...